### PR TITLE
Add support for bool literals in Julia

### PR DESCRIPTION
**Pull request by axic**
Depends on #2222.

### CODE REVIEW

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
There are compiler errors.

File: libsolidity/inlineasm/AsmPrinter.cpp [hunk 1]
```diff
@@ -46,8 +46,16 @@ string AsmPrinter::operator()(assembly::Instruction const& _instruction)
 
 string AsmPrinter::operator()(assembly::Literal const& _literal)
 {
-	if (_literal.isNumber)
+	switch(_literal.kind)
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmPrinter.cpp]: Please add a space after `switch`.

File: libsolidity/inlineasm/AsmParser.cpp [hunk 2]
```diff
@@ -206,10 +206,27 @@ assembly::Statement Parser::parseElementaryOperation(bool _onlySinglePusher)
 	}
 	case Token::StringLiteral:
 	case Token::Number:
+	case Token::TrueLiteral:
+	case Token::FalseLiteral:
 	{
+		LiteralKind kind;
+		switch(m_scanner->currentToken())
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmParser.cpp]: Please add a space after `switch`.

**Comment by axic** [comment]
This needs tests too.

File: libsolidity/inlineasm/AsmCodeGen.cpp [hunk 3]
```diff
@@ -118,8 +118,15 @@ class CodeTransform: public boost::static_visitor<>
 	void operator()(assembly::Literal const& _literal)
 	{
 		m_state.assembly.setSourceLocation(_literal.location);
-		if (_literal.isNumber)
+		if (_literal.kind == assembly::LiteralKind::String)
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmCodeGen.cpp]: Should read `assembly::LiteralKind::Number`.

**Comment by axic** [comment]
@chriseth updated, this should pass now

**Pull request merged**